lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -849,6 +849,98 @@ llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     return llvm::Error::success();
   }
 
+  if (typeName == "SparseLengthsSumFused8BitRowwise" ||
+      typeName == "SparseLengthsWeightedSumFused8BitRowwise") {
+    // If SparseLengthsWeightedSumFused8BitRowwise, then the weights are the
+    // second input and so we need to shift indices/lengths.
+    size_t indicesIdx = 1;
+    size_t lengthsIdx = 2;
+    if (typeName == "SparseLengthsWeightedSumFused8BitRowwise") {
+      indicesIdx++;
+      lengthsIdx++;
+    }
+
+    NodeValue data;
+    ASSIGN_VALUE_OR_RETURN_ERR(data,
+                               getNodeValueOrCreateConstantByName(op.input(0)));
+    NodeValue indices;
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        indices, getNodeValueOrCreateConstantByName(op.input(indicesIdx)));
+    NodeValue lengths;
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        lengths, getNodeValueOrCreateConstantByName(op.input(lengthsIdx)));
+
+    // Each row of data stores quantized values, and then 4-byte scale and
+    // 4-byte bias. Thus data must be a Constant so we can retrieve the
+    // Scales/Offsets inputs for RowwiseQuantizedSparseLengthsWeightedSumNode.
+    Constant *fusedDataC = llvm::dyn_cast<Constant>(data);
+    RETURN_ERR_IF_NOT(
+        fusedDataC,
+        "Input data must be Constant in order to retrieve Scales/Offsets.");
+    RETURN_ERR_IF_NOT(fusedDataC->getElementType() == ElemKind::Int8QTy,
+                      "Data must be Int8QTy.");
+    auto dataDims = fusedDataC->dims();
+    static_assert(sizeof(float) == 4, "float must be 4 bytes.");
+    RETURN_ERR_IF_NOT(dataDims.size() == 2 && dataDims[1] > 8,
+                      "Fused data must be 2 dimemsional, with 8 extra columns "
+                      "for holding the per row float scale and offset");
+
+    // Now we strip out the data and the scales/offsets and put them in their
+    // own Constants. Note that we use 0.0/0 for the scale/offset of rwqData, as
+    // the actual scales/offsets are provided in dataScales/dataOffsets.
+    const size_t secondDimDataLen = dataDims[1] - 8;
+    Constant *rwqData = G_.getParent()->createConstant(
+        ElemKind::Int8QTy, {dataDims[0], secondDimDataLen}, 0.0, 0, "data");
+    Constant *dataScales = G_.getParent()->createConstant(
+        ElemKind::FloatTy, {dataDims[0]}, "dataScales");
+    Constant *dataOffsets = G_.getParent()->createConstant(
+        ElemKind::Int32ITy, {dataDims[0]}, "dataOffsets");
+
+    // Now strip out all the data from the fused Constant and place them
+    // correctly in their respective Constant payloads. First set the rwqData
+    // from fusedData, skipping over the scales/offsets.
+    auto fusedDataH = fusedDataC->getHandle<int8_t>();
+    auto rwqDataH = rwqData->getHandle<int8_t>();
+    for (size_t i = 0; i < dataDims[0]; i++) {
+      for (size_t j = 0; j < secondDimDataLen; j++) {
+        rwqDataH.at({i, j}) = fusedDataH.at({i, j});
```

Comment:
is fused data stored as signed int8?

Reply:
At this point in Glow, yes -- Caffe2 uses uint8, but we convert it to int8 when we load from `Int8GivenTensorFill` (prior to this point).